src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitSuperClassTest.java

```diff
@@ -0,0 +1,169 @@
+        "com.puppycrawl.tools.checkstyle.checks.coding.SuperCloneCheck",
+        "com.puppycrawl.tools.checkstyle.checks.coding.SuperFinalizeCheck",
+        "com.puppycrawl.tools.checkstyle.checks.header.HeaderCheck",
+        "com.puppycrawl.tools.checkstyle.checks.header.RegexpHeaderCheck",
+        "com.puppycrawl.tools.checkstyle.checks.metrics.ClassDataAbstractionCouplingCheck",
+        "com.puppycrawl.tools.checkstyle.checks.metrics.ClassFanOutComplexityCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.AbstractAccessControlNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.CatchParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.ClassTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.IllegalIdentifierNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.InterfaceTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LambdaParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LocalFinalVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LocalVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MemberNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MethodTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.ParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.PatternVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.RecordComponentNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.RecordTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.StaticVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.TypeNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck",
+        "com.puppycrawl.tools.checkstyle.checks.whitespace.TypecastParenPadCheck"
+    );
+
+    /**
+     * ArchCondition checking that a class has a particular super class.
+     *
+     * @param superclass the superclass
+     * @return ArchCondition checking that a class has a particular super class
+     */
+    private static ArchCondition<JavaClass> haveSuperclass(Class<?> superclass) {
+        return new SuperclassArchCondition(superclass);
+    }
+
+    /**
+     * Tests that all checks have {@link AbstractCheck} or {@link AbstractFileSetCheck} or
+     * {@link AbstractJavadocCheck} as their super class.
+     */
+    @Test
+    public void testChecksShouldHaveAllowedAbstractClassAsSuperclass() {
+        final JavaClasses checksPackage = new ClassFileImporter()
+            .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_TESTS)
+            .importPackages("com.puppycrawl.tools.checkstyle.checks")
```

Comment:
Same comment as in other PR that we are missing some "checks" but specifying package.

Maybe I missed conversation, but why aren't we applying this to the filters, and such?

Reply:
We initially discussed checks to extend `AbstractCheck` and ......
If we create a rule for filters, etc, I think that should be a different PR. Can we go forward with:
```java
return ModuleReflectionUtil.isValidCheckstyleClass(clazz)       
    && (ModuleReflectionUtil.isCheckstyleTreeWalkerCheck(clazz) 
    || ModuleReflectionUtil.isFileSetModule(clazz));            
```

Reply:
Yes, and I am fine with another PR for other modules, but it seems like it would be rewriting this from checks to modules with the changes.

Reply:
Agreed, logic will be almost the same, but we can discuss it separately.